### PR TITLE
[google_maps_flutter] Android Code Inspection and Clean up

### PR DESCRIPTION
**Pull request by hamdikahloun**
## Description

- Handle deprecation & unchecked warning as error

- Avoiding uses or overrides a deprecated API

## Related Issues

flutter/flutter#65970

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapBuilder.java [hunk 1]
```diff
@@ -11,8 +11,7 @@
 import com.google.android.gms.maps.GoogleMapOptions;
 import com.google.android.gms.maps.model.CameraPosition;
 import com.google.android.gms.maps.model.LatLngBounds;
-import io.flutter.plugin.common.BinaryMessenger;
-import io.flutter.plugin.common.PluginRegistry;
+import io.flutter.plugin.common.*;
```
Comment (dnfield) [inline packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapBuilder.java:14]: Please use specific imports, avoid *

File: packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java [hunk 2]
```diff
@@ -42,10 +42,7 @@
 import com.google.android.gms.maps.model.Polygon;
 import com.google.android.gms.maps.model.Polyline;
 import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding;
-import io.flutter.plugin.common.BinaryMessenger;
-import io.flutter.plugin.common.MethodCall;
-import io.flutter.plugin.common.MethodChannel;
-import io.flutter.plugin.common.PluginRegistry;
+import io.flutter.plugin.common.*;
```
Comment (dnfield) [inline packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java:45]: Ditto

Comment (hamdikahloun) [reply packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java:45]: i have this warning if i just import PluginRegistry : 

```
warning: [deprecation] PluginRegistry in io.flutter.plugin.common has been deprecated
import io.flutter.plugin.common.PluginRegistry;
```

Comment (dnfield) [reply packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java:45]: @xster @amirh @blasten - do any of you know if the maps plugin still needs deprecated functionality from PluginRegistry? I have some vague memory of it being special for some reason.

Comment (hamdikahloun) [reply packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java]: @dnfield i fixed using  `io.flutter.plugin.common.PluginRegistry.Registrar`

File: packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapBuilder.java [hunk 3]
```diff
@@ -36,7 +35,7 @@ GoogleMapController build(
       BinaryMessenger binaryMessenger,
       Application application,
       Lifecycle lifecycle,
-      PluginRegistry.Registrar registrar,
+      @SuppressWarnings("deprecation") io.flutter.plugin.common.PluginRegistry.Registrar registrar,
```
Comment (dnfield) [inline packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapBuilder.java:38]: @cyanglaz or @blasten what's the plan for this longer term? I remember something funky about this plugin and the v2 embedding but not specifics.

Comment (hamdikahloun) [reply packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapBuilder.java]: @dnfield it's fixed

File: packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java (hunk 2 above)
Comment (xster) [reply packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java]: Correct, you can't import deprecated classes. You have to use it by the fully qualified name and add the lint ignore statement on the same line.

File: packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/Convert.java [hunk 4]
```diff
@@ -45,15 +45,21 @@ private static BitmapDescriptor toBitmapDescriptor(Object o) {
       case "fromAsset":
         if (data.size() == 2) {
           return BitmapDescriptorFactory.fromAsset(
-              FlutterMain.getLookupKeyForAsset(toString(data.get(1))));
+              FlutterInjector.instance()
```
Comment (xster) [inline packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/Convert.java:48]: nit, just keep a local reference to the flutter loader to keep it shorter

**Review by xster — COMMENTED** [review]
The API dedeprecation usage LGTM

File: packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/Convert.java [hunk 5]
```diff
@@ -34,6 +35,7 @@
 class Convert {
 
   private static BitmapDescriptor toBitmapDescriptor(Object o) {
+    final FlutterLoader flutterLoader = FlutterInjector.instance().flutterLoader();
```
Comment (dnfield) [inline packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/Convert.java:38]: This probably isn't present on stable, or if it is, we have to increase the minimum version Flutter SDK constraint most likelyt o capture it. /cc @xster

Comment (dnfield) [reply packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/Convert.java:38]: We should also consider whether we just ignore a depprecation warning for this or not - xster will have more context there.

**Review by dnfield — CHANGES_REQUESTED** [review]
Need to figure out the Flutter SDK constraint or used deprecated API

File: packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/Convert.java (hunk 5 above)
Comment (xster) [reply packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/Convert.java:38]: phew, glad you caught me. Forgot about targeting stable. We should indeed leave an ignore like https://github.com/flutter/plugins/pull/3072/files#diff-e57a1a7ffb729ad174b597626202429c1c87e4168e19b2aa9decc1be84e8777bR56 with a TODO to remove when the next stable ships. For now we have to use the old APIs.

File: packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/Convert.java [hunk 6]
```diff
@@ -33,6 +33,9 @@
 /** Conversions between JSON-like values and GoogleMaps data types. */
 class Convert {
 
+  // TODO(hamdikahloun): FlutterMain has been deprecated and should be replaced with FlutterLoader
+  //  when it's available in Stable channel.
```
Comment (dnfield) [inline packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/Convert.java:37]: Please file an issue if there isn't one already and link it here.

**Review by dnfield — APPROVED** [review]
LGTM modulo filing an issue for the TODO

**Pull request merged**